### ISSUE TITLE
RS discovery of forgerock aspsp does not serialise to OBDiscoveryResponse

### ISSUE THREAD

**Issue by Thomas-Franklin** [OpenBankingToolkit/openbanking-uk-datamodel#16]
**Describe the bug**

When calling the forgerock aspsp discovery endpoint and trying to map response to `OBDiscoveryResponse` it fails as the `@type` ID is not valid for the mapping in `OBDiscoveryAPILinks`

**To Reproduce**
Steps to reproduce the behaviour:

https://github.com/Thomas-Franklin/ob-discovery-response-bug

Above repo is minimum setup to reproduce;

```
./gradlew clean build
java -jar build/libs/demo-0.0.1-SNAPSHOT.jar\
```

In another terminal perform:

```
curl --location --request GET 'http://localhost:8080/discovery?discoveryUrl=https://<forgerock-matls-rs-url>/open-banking/discovery'
```

**Expected behaviour**

The `OBDiscoveryResponse` class should be able to map to the response body from the forgerock discovery endpoint.

**Current behaviour**
A clear and concise description of what the current behaviour.

Currently the class fails serialisation as the `@type` in the forgerock discovery endpoint as it listed as `GenericOBDiscoveryAPILinks` but the toolkit (latest version) has the supported types as:

```
@JsonSubTypes({
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksAccount1.class, name = "OBDiscoveryAPILinksAccount1"),
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksAccount2.class, name = "OBDiscoveryAPILinksAccount2"),
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksAccount3.class, name = "OBDiscoveryAPILinksAccount3"),
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksPayment1.class, name = "OBDiscoveryAPILinksPayment1"),
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksPayment3.class, name = "OBDiscoveryAPILinksPayment3"),
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksPayment3.class, name = "OBDiscoveryAPILinksPayment4"),
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksFundsConfirmation3.class, name = "OBDiscoveryAPILinksFundsConfirmation3"),
        @JsonSubTypes.Type(value = OBDiscoveryAPILinksEventNotification3.class, name = "OBDiscoveryAPILinksEventNotification3"),
})
```

Which means the mapping fails, the stacktrace can be seen in the repo above.

**Additional context**
OPTIONAL - Add any other context about the problem here.

Be nice to have a like for like environment before switching to actual aspsps (as we're still early in our development and not ready to test against actual aspsps)

**Code analysis**

Add an `GenericOBDiscoveryAPILinks` class as a subtype to support the forgerock discovery endpoint, or update the API to return the types expected of an actual ASPSP

**Comment by jorgesanchezperez** [comment]
Thanks @Thomas-Franklin to report the Issue. A team member will check it.

**Comment by jorgesanchezperez** [comment]
Hi @Thomas-Franklin this bug has been fixed since release version [3.1.2.25](https://github.com/OpenBankingToolkit/openbanking-uk-datamodel/releases/tag/3.1.2.25).
I recommend you to use the latests release version.
- Current release version [v3.1.6.0](https://github.com/OpenBankingToolkit/openbanking-uk-datamodel/releases/tag/3.1.6.0)

**Comment by matt-wills** [comment]
Additionally to Jorge's comment above (which addresses the typo in the mapping of `OBDiscoveryAPILinksPayment4`), v3.1.6.1 of openingbanking-uk-datamodel includes `GenericOBDiscoveryAPILinks`.
